### ISSUE TITLE
Deploying preview of site for PRs

### ISSUE THREAD

**Issue by oxinabox** [tlienart/Franklin.jl#787]
This is basically the same as https://github.com/JuliaDocs/Documenter.jl/issues/1131

It would be awesome if we had an option for PRs to run CI and push to some URL so you can eyeball some changes
Documentor does `[normalurl]/previews/PR[number]` e.g. https://juliadiff.org/ChainRulesCore.jl/previews/PR287/

**Comment by tlienart** [comment]
Yes so the same stuff with a `/preview/` URL is possible (and could be discarded upon merging into the main branch). I'll look into it.

In the meantime this can be done with a (free) netlify hook; this is what's done with the julialang website: https://github.com/JuliaLang/www.julialang.org/blob/main/.github/workflows/pr_comment.yml

**Comment by oxinabox** [comment]
I would love a way to do it from untrusted branches, e.g. PRs from forks.
which does suggest *not* merging into the main gh-pages branch in a folder.
and rather sending it somewhere of its own with a clearly unofficial URL

**Comment by tlienart** [comment]
for the "clearly unofficial URL" I think the netlify option is a good one; you can reuse the `pr_comment` script from above which deploys to a `preview` branch that gets picked up by netlify and is deployed at something like `somerandomaddress.netlify.app/pr/123` 

the alternative (which is what I thought you were suggesting) is to have something like a `__pr__123` folder inside the `__site/` folder on `gh-pages` that gets filled with the content of the site generated by the PR, and that it then becomes  available at `yourbaseurl.io/__pr__123/` ; upon eventual merge into the master branch, these `__pr__xxx` folders would get removed.

Wdyt?

**Comment by oxinabox** [comment]
> the alternative (which is what I thought you were suggesting) is

I don't really have opinions on how it should work.
I just want the GHA to be setup by default (potentially commented out), and instructions on what to do (if anything) in the docs.
